src/goto-programs/abstract-interpretation/interval_domain.cpp

```diff
@@ -1091,8 +1089,12 @@ bool interval_domaint::join(const interval_domaint &b)
     return true;
   }
 
-  bool result = join(int_map, b.int_map) || join(real_map, b.real_map) ||
-                join(wrap_map, b.wrap_map);
+  const bool is_guard = to->is_assume() || to->is_assert() ||
+                        (to->is_goto() && !is_true(to->guard));
```

Comment:
Why do you need this `!is_true(to->guard)`? It isn't very clear...

Reply:
Thats to differentiate IF GOTO from plain GOTO.

Reply:
Can you please add this explanation as a comment? :-)

Reply:
I added a new intermediate variable